components/type_info.rst

```diff
@@ -0,0 +1,59 @@
+The TypeInfo Component
+======================
+
+    The TypeInfo component extracts PHP types information. It aims to:
+
+    - Have a powerful Type definition that can handle union, intersections, and generics (and could be even more extended)
+
+    - Being able to get types from anything, such as properties, method arguments, return types, and raw strings (and can also be extended).
+
+
+Installation
+------------
+
+.. code-block:: terminal
+
+    $ composer require symfony/type-info
+
+.. include:: /components/require_autoload.rst.inc
+
+Usage
+-----
+
+There is two ways to use this component. First one is to create a manually thanks
+to :class:`Symfony\\Component\\TypeInfo\\Type` static methods as following::
+
+    use Symfony\Component\TypeInfo\Type;
+
+    Type::int();
```

Comment:
I don't understand the examples of this block. What are we building? Is this to create something? To check something? How can I use this in a Symfony app? Thanks.

Reply:
This is not a component you will use in a Symfony app. This is a component made for other components or for libraries to use.